### ISSUE TITLE
datatips only with modifier

### ISSUE THREAD

**Issue by chrisdrackett** [facebook-atom/atom-ide-ui#80]
I'm really loving the functionality provided by datatips... when I'm looking for that kind of information. Sadly a lot of the time the tip (or the highlight that comes with it) gets in the way of the work I'm doing. I've played with the delay values, which can help, but I've also been thinking it might be really awesome to have an option for datatips to only show up when a modifier (say command or similar) is pressed. This would make they quick to access when needed, but keep them out of the way when not.

**Comment by Gert-dev** [comment]
Agreed. It seems currently a provider can already indicate that a datatip only should be shown when a specified modifier is pressed (via `heldKeys`), but this is mostly only interesting if the provider actually wants to determine what key should be held down.

In other words, for run-of-the-mill datatips it would be handy if a user could indicate that datatips should trigger only when the configured hyperclick modifier is pressed. Without this, the provider would somehow have to determine what the user's hyperclick modifier key is and match that to `heldKeys` itself, which means fetching and parsing another package's settings.

Furthermore, the hyperclick modifier seems ideal for this, as you usually automatically get an underlining for symbols anyway, after which the datatip will pop up. Because the datatip only activates after a short timeout, it should, at least by default, not bother users that are just using the modifier key to click through (and navigate).

Finally, this moves the activation of the datatip to an actual user trigger, rather than undesirably popping up when leaving the mouse over an element by accident.

**Comment by chrisdrackett** [comment]
never-mind that the text highlight is the same as the selected text highlight, making selecting text within a object with a tip especially challenging
